ktor-http/common/src/io/ktor/http/header/utils.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2014-2025 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.http.header
+
+import io.ktor.http.*
+
+/**
+ * Set of headers that must not be treated as comma-separated lists.
+ *
+ * These headers are single-valued by grammar or allow commas within a single value.
+ * For them, values are returned exactly as received—one element per field-line—with no comma splitting.
+ */
+internal val NON_LIST_HEADERS: Set<String> = setOf(
+    HttpHeaders.SetCookie,
+    HttpHeaders.Cookie,
+    HttpHeaders.Authorization,
+    HttpHeaders.ProxyAuthorization,
+    HttpHeaders.ETag,
+    HttpHeaders.IfRange,
+    HttpHeaders.Date,
+    HttpHeaders.LastModified,
+    HttpHeaders.Expires,
+    HttpHeaders.RetryAfter,
+    HttpHeaders.ContentDisposition,
+    HttpHeaders.Location,
+    HttpHeaders.ContentLocation,
+).map { it.lowercase() }.toSet()
+
+/**
+ * Returns `true` when [name] should be treated as a comma-separated list header.
+ */
+internal fun isCommaSeparatedHeader(name: String): Boolean = name.lowercase() !in NON_LIST_HEADERS
+
+/**
+ * Splits header's values on commas when the header is treated as a comma-separated list.
+ */
+internal fun splitValuesForHeader(values: Map<String, List<String>>) = values.entries.associate { (k, v) ->
```

Comment:
Please also consider specifying the type explicitly. It would also be nice to have separate method names. `splitValueorHeader` has 2 overloads with different complexity